pyproject.toml

```diff
@@ -1,10 +1,20 @@
 [tool.ruff]
 line-length = 120
-target-version = "py311"
+target-version = "py312"
+lint.select = ["E", "F", "I", "UP"]
+lint.ignore = ["D203"]
+lint.isort.known-first-party = ["fastapi_app"]
 
-[tool.ruff.lint]
-select = ["E", "F", "I", "UP"]
-ignore = ["D203"]
+[tool.mypy]
+check_untyped_defs = true
+python_version = 3.12
+exclude = [".venv/*"]
```

Comment:
I ran mypy on the whole thing the only things left are the missing module stubs.
Don't know if you wanna do `#type: ignore` or add them to `pyproject.toml`